packages/garbo/src/tasks/barfTurn.ts

```diff
@@ -340,6 +343,61 @@ function aprilingSaxophoneLucky(additionalReady: () => boolean) {
   };
 }
 
+function augustScepterLucky(additionalReady: () => boolean) {
+  return {
+    completed: () =>
+      !shouldAugustCast($skill`Aug. 2nd: Find an Eleven-Leaf Clover Day`),
+    ready: () =>
+      additionalReady() &&
+      getBestLuckyAdventure().phase === "barf" &&
+      getBestLuckyAdventure().value() > get("valueOfAdventure"),
+    do: () => getBestLuckyAdventure().location,
+    prepare: () => {
+      if (!have($effect`Lucky!`)) {
+        useSkill($skill`Aug. 2nd: Find an Eleven-Leaf Clover Day`);
+        if (!have($effect`Lucky!`)) {
+          set("_aug2Cast", true);
+        }
+      }
+    },
+    combat: new GarboStrategy(() =>
+      Macro.abortWithMsg("Unexpected combat while attempting Lucky! adventure"),
+    ),
+    turns: () =>
+      shouldAugustCast($skill`Aug. 2nd: Find an Eleven-Leaf Clover Day`)
+        ? 1
+        : 0,
+    spendsTurn: true,
+  };
+}
+
+function pillKeeperLucky(additionalReady: () => boolean) {
+  return {
+    completed: () => get("_freePillKeeperUsed"),
+    ready: () =>
+      additionalReady() &&
+      getBestLuckyAdventure().phase === "barf" &&
+      getBestLuckyAdventure().value() > get("valueOfAdventure") &&
+      have($item`Eight Days a Week Pill Keeper`),
+    do: () => getBestLuckyAdventure().location,
+    prepare: () => {
+      if (!have($effect`Lucky!`)) {
+        retrieveItem($item`Eight Days a Week Pill Keeper`);
+        cliExecute("pillkeeper semirare");
+        if (!have($effect`Lucky!`)) {
+          set("_freePillKeeperUsed", true);
+          return;
+        }
+      }
+    },
```

Comment:
I think we should use a system similar to the one you've set up in embezzler.ts and for similar reasons (that is, separate lucky source from lucky execution).

BUT, the `turns` field should remain on the sources rather than on the lucky adventure task.